src/main/java/net/glowstone/block/state/StateSerialization.java

```diff
@@ -0,0 +1,85 @@
+package net.glowstone.block.state;
+
+import net.glowstone.block.state.impl.WoolStateDataReader;
+import org.bukkit.DyeColor;
+import org.bukkit.Material;
+import org.bukkit.material.MaterialData;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StateSerialization {
+
+    private static final Map<Material, BlockStateReader> READERS = new HashMap<>();
+
+    public static BlockStateData parse(Material material, String state) throws InvalidBlockStateException {
+        if (state == null || state.trim().isEmpty() || state.trim().equals("*")) {
+            return new BlockStateData();
+        }
+        BlockStateReader reader = getReader(material);
+        if (material == null || reader == null) {
+            throw new InvalidBlockStateException(material, state);
+        }
+        BlockStateData data = new BlockStateData();
+        String[] split = state.trim().split(",");
```

Comment:
I hadn't thought of using a regex here, I will consider it. My only concern would be readability, but that can be resolved with a comment.

Reply:
After checking, the regular expression you provided doesn't work all the time. For the string `a=b,c=d,e=f`:

Full match | `a=b,c=d,e=f`
-- | --
Group 1 | `a=b`
Group 2 | `a`
Group 3 | `b`
Group 4 | `,e=f`
Group 5 | `e`
Group 6 | `f`

This would skip the middle section.